pkg/rke2/spw.go

```diff
@@ -42,57 +26,55 @@ func checkStaticManifests(dataDir string) cmds.StartupHook {
 	return func(ctx context.Context, wg *sync.WaitGroup, args cmds.StartupHookArgs) error {
 		go func() {
 			defer wg.Done()
+			if err := wait.PollImmediate(20*time.Second, 30*time.Minute, func() (bool, error) {
 
-			var conn *grpc.ClientConn
-			if err := wait.ExponentialBackoff(criBackoff, func() (done bool, err error) {
-				conn, err = containerdk3s.CriConnection(ctx, containerdSock)
+				conn, err := containerdk3s.CriConnection(ctx, containerdSock)
 				if err != nil {
 					logrus.Infof("Waiting for cri connection: %v", err)
 					return false, nil
 				}
-				return true, nil
-			}); err != nil {
-				logrus.Fatalf("failed to setup cri connection: %v", err)
-			}
-			cRuntime := runtimeapi.NewRuntimeServiceClient(conn)
-			defer conn.Close()
+				cRuntime := runtimeapi.NewRuntimeServiceClient(conn)
+				defer conn.Close()
 
-			manifestDir := podManifestsDir(dataDir)
+				manifestDir := podManifestsDir(dataDir)
 
-			for _, pod := range []string{"etcd", "kube-apiserver"} {
-				manifestFile := filepath.Join(manifestDir, pod+".yaml")
-				if f, err := os.Open(manifestFile); err == nil {
-					podManifest := v1.Pod{}
-					decoder := yaml.NewYAMLToJSONDecoder(f)
-					err = decoder.Decode(&podManifest)
-					if err != nil {
-						logrus.Fatalf("Failed to decode %s manifest: %v", pod, err)
-					}
-					podFilter := &runtimeapi.ContainerFilter{
-						LabelSelector: map[string]string{
-							"io.kubernetes.container.name": pod,
-						},
-					}
-					if err := wait.ExponentialBackoff(podCheckBackoff, func() (done bool, err error) {
+				for _, pod := range []string{"etcd", "kube-apiserver"} {
+					manifestFile := filepath.Join(manifestDir, pod+".yaml")
+					if f, err := os.Open(manifestFile); err == nil {
+						podManifest := v1.Pod{}
+						decoder := yaml.NewYAMLToJSONDecoder(f)
+						err = decoder.Decode(&podManifest)
+						if err != nil {
+							logrus.Fatalf("Failed to decode %s manifest: %v", pod, err)
+						}
+						podFilter := &runtimeapi.ContainerFilter{
+							LabelSelector: map[string]string{
+								"io.kubernetes.container.name": pod,
```

Comment:
This will cause problems if someone ever deploys a pod with a container named 'etcd' or 'kube-apiserver. We should just filter on the pod UID we're looking for:
```suggestion
								"io.kubernetes.pod.uid": podManifest.UID,
```